### ISSUE TITLE
NotImplementedError: ibf_dump_object_unsupported occured when >=v0.4.0

### ISSUE THREAD

**Issue by st6446** [kachick/ruby-ulid#233]
here is reproduce code
https://github.com/st6446/temp_rails_app/commits/main

-----

## Backgroud
When I updated ruby-ulid `0.3.0` to `>=0.4.0`, some strange error occured.
```
bundle exec rspec spec/requests/graphql/xxx/yyy_spec.rb

An error occurred while loading ./spec/requests/graphql/xxx/yyy_spec.rb.
Failure/Error: require File.expand_path('../config/environment', __dir__)

NotImplementedError:
  ibf_dump_object_unsupported: 0x00007f1735c8a3e0 [3LM    ] T_ICLASS 
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/compile_cache/iseq.rb:42:in `to_binary'
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/compile_cache/iseq.rb:42:in `input_to_storage'
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/compile_cache/iseq.rb:60:in `fetch'
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/compile_cache/iseq.rb:60:in `fetch'
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/compile_cache/iseq.rb:85:in `load_iseq'
# /vendor/bundle/ruby/3.1.0/gems/ruby-ulid-0.4.0/lib/ruby-ulid.rb:7:in `require_relative'
# /vendor/bundle/ruby/3.1.0/gems/ruby-ulid-0.4.0/lib/ruby-ulid.rb:7:in `<main>'
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
# /vendor/bundle/ruby/3.1.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
# /usr/local/bundle/gems/bundler-2.3.19/lib/bundler.rb:188:in `require'
# ./config/application.rb:21:in `<top (required)>'
# ./config/environment.rb:4:in `require_relative'
# ./config/environment.rb:4:in `<top (required)>'
# ./spec/rails_helper.rb:7:in `require'
# ./spec/rails_helper.rb:7:in `<top (required)>'
# ./spec/requests/graphql/xxx/yyy_spec.rb:3:in `require'
# ./spec/requests/graphql/xxx/yyy_spec.rb:3:in `<top (required)>'
No examples found.


Finished in 0.00005 seconds (files took 0.59508 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```

Only I can resolve this problem by disabling bootsnap.
```
DISABLE_BOOTSNAP=true bundle exec rspec spec/requests/graphql/xxx/yyy_spec.rb
```

**Comment by kachick** [comment]
Thank you for the report! It sounds Ractor and/or bootsnap issue 🤔 

https://github.com/Shopify/bootsnap/issues/376 and https://bugs.ruby-lang.org/issues/18232 might be related...?

Which ruby version do you use? Could you share your ruby version with `ruby -v`?
And please ensure it is latest patched version. 🙏

And so, this gem also provide 0.6.1 for the latest. So updating to it makes my happy!

**Comment by kachick** [comment]
```console
❯ ruby -v -e "p RubyVM::InstructionSequence.compile_file('./lib/ruby-ulid.rb').to_binary; require_relative 'lib/ulid/version'; p ULID::VERSION"
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin21]
"YARB\x03\x00\x00\x00\x01\x00\x00\x004\x01\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\a\x00\x00\x00\x98\x00\x00\x00\x18\x01\x00\x00x86_64-darwin21\x00#%\acu\x0F\x01\x03\x0F\x03\x01\x0F\x01\x01\x0F\x00\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\x01\x01\x03\x05\x05\x00\x00\x00\x00\x00\x00\x00\t)\x03\x01\x01\x01\rS\v\x01\x01\x01\x01\x01\x01\x01\x01I\x01\x03\x05\x05\a\a\x03\x01\x0F1I!\t\v\x01\v\x00\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\x01\x00\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\v\x03\x01\x01\x01\x03\x05\x01\x01\x00\x00]\x00\x00\x00\xF1\x11\x00\x00G\x05\v\rE\x05\r<main>\x00\x00\x00E\x05\tulid\x00\x14\x05!require_relative\x00E\x03%./lib/ruby-ulid.rb\x00\x00\x00E\x03_/Users/kachick/repos/ruby-ulid/lib/ruby-ulid.rb\x00\x00\x9C\x00\x00\x00\xA0\x00\x00\x00\xA4\x00\x00\x00\xB0\x00\x00\x00\xB8\x00\x00\x00\xCC\x00\x00\x00\xE4\x00\x00\x00"
"0.4.0"
```

Looks working in pure ruby layer 🤔

**Comment by kachick** [comment]
```
❯ ruby -v -e "p RubyVM::InstructionSequence.compile_file('./lib/ulid.rb').to_binary; require './lib/ulid/version.rb'; p ULID::VERSION"
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin21]
-e:1:in `to_binary': ibf_dump_object_unsupported: 0x000000010e1b3438 [3LM    ] T_ICLASS  (NotImplementedError)
	from -e:1:in `<main>'
```

Reproduced! I'll dig 🙇

**Comment by kachick** [comment]
📝 It is a bit curios behavior. Declaring shareable_constant_value pragma and assigning constants with method calling raising this error (even if Object#freeze).  🙄

**Comment by kachick** [comment]
I have fixed RubyVM::InstructionSequence compatible issue in #235 and released in new version 0.7.0. Could you check new version will pass in your environment?
